### PR TITLE
Publish to Hex.pm from Github Action

### PR DESCRIPTION
**Pull request by andyduong1920**
## What happened

Solved https://github.com/nimblehq/elixir-templates/issues/47
 
## Insight

Publish to Hex.pm from Github Action 
 
## Proof Of Work

image

image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -94,6 +94,14 @@ Putting it all together, it is 6 variant test cases.
 - Applying the `Web variant` to a `Standard Web project`
 - Applying the `Web variant` to a `Custom Web project`
 
+### Release
+
+The release process is following the [Git flow](https://nimblehq.co/compass/development/version-control/#releases-).
+
+Once a `release/<version number>` is created, to publish the new version to https://hex.pm/packages/nimble_phx_gen_template, we need to increase the version number in the `mix.ex` file on the release branch before the merge action.
+
+Once the release branch is merged into the master, Github Action will publish the template to https://hex.pm/packages/nimble_phx_gen_template automatically.
```
Comment (hanam1ni) [inline README.md:103]:
```suggestion
Once the release branch is merged into the master, Github Action will automatically publish the template to https://hex.pm/packages/nimble_phx_gen_template.
```

File: README.md [hunk 2]
```diff
@@ -94,6 +94,14 @@ Putting it all together, it is 6 variant test cases.
 - Applying the `Web variant` to a `Standard Web project`
 - Applying the `Web variant` to a `Custom Web project`
 
+### Release
+
+The release process is following the [Git flow](https://nimblehq.co/compass/development/version-control/#releases-).
+
+Once a `release/<version number>` is created, to publish the new version to https://hex.pm/packages/nimble_phx_gen_template, we need to increase the version number in the `mix.ex` file on the release branch before the merge action.
```
Comment (hanam1ni) [inline README.md:101]:
```suggestion
Once a `release/<version number>` is created, to publish the new version to Hex.pm, the version number in the `mix.ex` file needs to be updated on the release branch before merging.
```

File: README.md [hunk 3]
```diff
@@ -94,6 +94,14 @@ Putting it all together, it is 6 variant test cases.
 - Applying the `Web variant` to a `Standard Web project`
 - Applying the `Web variant` to a `Custom Web project`
 
+### Release
```
Comment (hanam1ni) [inline README.md:97]: How about we document that the `HEX_API_KEY` secret needs to be set before making a release?

**Review by hanam1ni — COMMENTED** [review]
LGTM, only minor comments in README 👍

File: README.md (hunk 1 above)
Comment (andyduong1920) [reply README.md]: Fixed in https://github.com/nimblehq/elixir-templates/pull/63/commits/421a87bde1c932dfd3301dde28b40f7a95a697be @hanam1ni

File: README.md (hunk 2 above)
Comment (andyduong1920) [reply README.md]: Fixed in https://github.com/nimblehq/elixir-templates/pull/63/commits/421a87bde1c932dfd3301dde28b40f7a95a697be @hanam1ni

File: README.md (hunk 3 above)
Comment (andyduong1920) [reply README.md:97]: Fixed in https://github.com/nimblehq/elixir-templates/pull/63/commits/421a87bde1c932dfd3301dde28b40f7a95a697be @hanam1ni

**Review by hanam1ni — APPROVED** [review]

File: .github/workflows/publish_to_hex_pm.yml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+      - completed
+  workflow_dispatch:
+
+env:
+  OTP_VERSION: 23.1.4
+  ELIXIR_VERSION: 1.11.2
+
+jobs:   
+  publish:
+    runs-on: ubuntu-latest
```
Comment (olivierobert) [inline .github/workflows/publish_to_hex_pm.yml:19]: When using `workflow_run`, you also need to add a conditional to also be able to re-run the job manually:

```
if: ${{ github.event_name != 'workflow_run' || github.event.workflow_run.conclusion == 'success' }}
```

File: README.md [hunk 5]
```diff
@@ -94,6 +94,16 @@ Putting it all together, it is 6 variant test cases.
...
+
+Set the `HEX_API_KEY` as a Github secret (skip this step if it has been done)
```
Comment (olivierobert) [inline README.md:99]: Missing period:

```suggestion
Set the `HEX_API_KEY` as a Github secret (skip this step if it has been done).
```

File: README.md [hunk 6]
```diff
@@ -94,6 +94,16 @@ Putting it all together, it is 6 variant test cases.
...
+
+The release process is following the [Git flow](https://nimblehq.co/compass/development/version-control/#releases-).
```
Comment (olivierobert) [inline README.md:101]: Tense:

```suggestion
The release process follows the [Git flow](https://nimblehq.co/compass/development/version-control/#releases-).
```

There is no need for the continuous present here as the process has been defined and is no longer active.

File: README.md [hunk 7]
```diff
@@ -94,6 +94,16 @@ Putting it all together, it is 6 variant test cases.
...
+
+Once a `release/<version number>` is created, to publish the new version to Hex.pm, the version number in the `mix.ex` file needs to be updated on the release branch before merging.
+
+Once the release branch is merged into the master, Github Action will automatically publish the template to https://hex.pm/packages/nimble_phx_gen_template.
```
Comment (olivierobert) [inline README.md:105]: Small clarification:

```suggestion
Once the release branch is merged into the `master` branch, Github Action will automatically publish the template to https://hex.pm/packages/nimble_phx_gen_template.
```

Comment (andyduong1920) [reply README.md]: Fixed in https://github.com/nimblehq/elixir-templates/pull/63/commits/b6cc5fc1aaf57f9bb3d280bbc73ce105c357b8bf @olivierobert

File: README.md (hunk 6 above)
Comment (andyduong1920) [reply README.md]: Fixed in https://github.com/nimblehq/elixir-templates/pull/63/commits/b6cc5fc1aaf57f9bb3d280bbc73ce105c357b8bf @olivierobert

File: README.md (hunk 5 above)
Comment (andyduong1920) [reply README.md]: Fixed in https://github.com/nimblehq/elixir-templates/pull/63/commits/b6cc5fc1aaf57f9bb3d280bbc73ce105c357b8bf @olivierobert

File: .github/workflows/publish_to_hex_pm.yml (hunk 4 above)
Comment (andyduong1920) [reply .github/workflows/publish_to_hex_pm.yml:19]: Good to know that @olivierobert, I have fixed that in https://github.com/nimblehq/elixir-templates/pull/63/commits/92806e395a01744416ee762ae04aaea7341720e0

**Review by bterone — APPROVED** [review]

**Review by olivierobert — APPROVED** [review]

**Pull request merged**